forum/startups/helpers.py

```diff
@@ -0,0 +1,78 @@
+from django.db.models import Q
+from .models import Startup
+from .serializers import StartupSerializer, StartupSizeSerializer
+
+from projects.models import Project, Industry
+from projects.serializers import ProjectSerializer, IndustrySerializer
+
+
+def select_startups_by_search_string(search_string):
+    return Startup.objects.filter(
+        Q(name__icontains = search_string) |
+        Q(location__icontains = search_string) |
+        Q(description__icontains = search_string)
+    )
+
+
+def filter_startups(query_params):
+    industry_name = query_params.get("industry")
+    max_budget = query_params.get("max-budget")
+    size = query_params.get("size")
+    
+    startups = Startup.objects.filter()
+
+    if industry_name:
+        industry = Industry.objects.filter(name=industry_name).first()
+        projects = Project.objects.filter(industries=industry)
+        startups_ids = [proj.startup.startup_id for proj in projects]
+        startups = startups.filter(pk__in=startups_ids)
+    
+    if max_budget:
+        try:
+            max_budget = int(max_budget)
+        except ValueError:
+            return []
```

Comment:
Try to use readable approach to return an empty queryset on invalid input for budget and size, e.g.
```
return Startup.objects.none()
```

Reply:
Changed